x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/user/TransportChangePasswordAction.java

```diff
@@ -62,6 +81,89 @@ protected void doExecute(Task task, ChangePasswordRequest request, ActionListene
+                // nativeUsersStore.changePassword can create a missing reserved user, so enter only if not reserved
+                if (ClientReservedRealm.isReserved(username, settings) == false && user == null) {
+                    List<Realm> nonNativeRealms = realms.getActiveRealms()
+                        .stream()
+                        .filter(t -> Set.of(NativeRealmSettings.TYPE, ReservedRealm.TYPE).contains(t.type()) == false) // Reserved realm is
+                                                                                                                       // implemented in the
+                                                                                                                       // native store
+                        .toList();
+                    if (nonNativeRealms.isEmpty()) {
+                        listener.onFailure(createUserNotFoundException());
+                        return;
+                    }
+
+                    GroupedActionListener<User> gal = new GroupedActionListener<>(nonNativeRealms.size(), new ActionListener<>() {
+                        @Override
+                        public void onResponse(Collection<User> users) {
+                            List<String> realmErrors = new ArrayList<>();
+                            for (User user : users) {
+                                if (user != null) {
+                                    realmErrors.add(
+                                        "user ["
+                                            + username
+                                            + "] does not belong to the native realm and cannot be"
+                                            + " managed via this API."
+                                    );
+                                    break;
+                                }
+                            }
+                            if (realmErrors.isEmpty()) {
+                                // user wasn't found in any other realm, display standard not-found message
+                                listener.onFailure(createUserNotFoundException());
+                            } else {
+                                ValidationException validationException = new ValidationException();
+                                realmErrors.forEach(validationException::addValidationError);
+                                listener.onFailure(validationException);
+                            }
+                        }
+
+                        @Override
+                        public void onFailure(Exception e) {
+                            listener.onFailure(e);
+                        }
+                    });
+                    for (Realm realm : nonNativeRealms) {
+                        EsExecutors.DIRECT_EXECUTOR_SERVICE.execute(
+                            ActionRunnable.wrap(gal, userActionListener -> realm.lookupUser(username, userActionListener))
```

Comment:
It seems that `MicrosoftGraphAuthzRealm` [always returns the non-null user](https://github.com/elastic/elasticsearch/blob/080a280345a4788b7682f4c5087381aaab64e4ed/plugins/microsoft-graph-authz/src/main/java/org/elasticsearch/xpack/security/authz/microsoft/MicrosoftGraphAuthzRealm.java#L45-L48). Not sure if this is intentional. If so, then we will always return the error (`"user [" + username + "] does not belong to the native realm and cannot be managed via this API."`) when Microsoft graph realm is enabled. This is not ideal as we should instead return "user not found". 

@jfreden could you helps us understand if always returning user in the lookup method is intentional?